content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -9,8 +9,186 @@ tocDepth: 3
 
 <Admonition type="warning">
 
-We plan to add `query` extensions to a future version of Prisma. This page is a placeholder.
+This page is in draft.
 
 </Admonition>
 
+From version 4.7.0, you can use `query` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) to create custom Prisma Client queries. These custom queries return type-safe data and, like all Prisma Client extensions, [run in isolated instances](/concepts/components/prisma-client/client-extensions#client-instances).
+
+You can use `query` extensions as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases), with the advantages of user isolation and end-to-end type safety. [Learn more](#extensions-versus-middlewares).
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. Use the `query` extension type to create custom queries to a specific model in your schema or to all models in your schema.
+
 </TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create custom queries
+
+To create a custom query, use the following structure.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  query?: {
+    user: { ... } //in this case, we add a query to the `user` model
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs.
+
+In the following example, a custom query finds users who are older than 18 years:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findMany({ model, action, args, data }) {
+        args.where.age = { gt: 18 }
```

Comment:
Since where is not always present, it can also be `undefined` which means `args.where.age` would fail in such scenario. The solution is to always assign an object, and backfill it with the other parts of the where clause of the current query.
```suggestion
        args.where = { age: { gt: 18 }, ...args.where }
```

Reply:
Thanks. I'll add a code comment summarizing your above point to explain why we've done it this way.